pkg/cfn/builder/vpc_endpoint.go

```diff
@@ -97,18 +101,43 @@ func (e *VPCEndpointResourceSet) subnetsForAZs(azs []string) []*gfnt.Value {
 	return subnetRefs
 }
 
-func (e *VPCEndpointResourceSet) routeTableIDs() []*gfnt.Value {
+func (e *VPCEndpointResourceSet) routeTableIDs() ([]*gfnt.Value, error) {
 	var routeTableIDs []*gfnt.Value
-	m := make(map[string]bool)
+	routeTables := make(map[string]bool)
+	routeTableDestinationPrefixListIDs := make(map[string]bool)
 	for _, subnet := range e.subnets {
 		routeTableStr := subnet.RouteTable.String()
 
-		if !m[routeTableStr] {
-			m[routeTableStr] = true
+		if !routeTables[routeTableStr] {
+			routeTables[routeTableStr] = true
+		} else {
+			continue
+		}
+
+		destinationExists := false
+		output, err := e.ec2API.DescribeRouteTables(&ec2.DescribeRouteTablesInput{
+			RouteTableIds: aws.StringSlice([]string{subnet.RouteTable.String()}),
```

Comment:
Uh, this has one weird case where this is 👍 

```
RouteTableIds: ["{map[Ref:PrivateRouteTableUSWEST2A]}"]
```

That will def not work. :/

Reply:
I probably will have to change all of this to properly pass in the routing table IDs because that's not the routing table ID. :sadpanda:

Reply:
Hmm, these should be IDs though.

```go
subnetRoutes[*rta.SubnetId] = *rt.RouteTableId
```

So the test must be wrong somewhere.